core/amber/src/main/resources/python_udf/topic_modeling_mallet_trainer.py

```diff
@@ -0,0 +1,90 @@
+import logging
+
+# Use gensim 3.8.3
+import gensim
+import gensim.corpora as corpora
+import pandas
+import os
+
+import texera_udf_operator_base
+
+
+class TopicModeling(texera_udf_operator_base.TexeraBlockingUnsupervisedTrainerOperator):
+    logger = logging.getLogger("PythonUDF.TopicModelingTrainer")
+
+    @texera_udf_operator_base.exception(logger)
+    def open(self, *args):
+        super(TopicModeling, self).open(*args)
+
+        # TODO: _train_args from user input args
+        if len(args) >= 3:
+            MALLET_HOME = str(args[1])
+            NUM_TOPICS = int(args[2])
+        else:
+            self.logger.exception("Not enough arguments in topic modeling operator")
+            raise RuntimeError("Not enough arguments in topic modeling operator")
+
+        MALLET_PATH = MALLET_HOME+"/bin/mallet"
+        RANDOM_SEED = 41
+        os.environ['MALLET_HOME'] = MALLET_HOME
+
+        self._train_args = {"mallet_path":MALLET_PATH, "random_seed":RANDOM_SEED, "num_topics":NUM_TOPICS}
+
+        self.logger.debug(f"getting args {args}")
+        self.logger.debug(f"parsed training args {self._train_args}")
+
+    @texera_udf_operator_base.exception(logger)
+    def accept(self, row: pandas.Series, nth_child: int = 0) -> None:
+        # override accept to accept rows as lists
+        self._data.append(row[0].strip().split())
+
+    @staticmethod
+    @texera_udf_operator_base.exception(logger)
+    def train(data, *args, **kwargs):
```

Comment:
as you are expecting `mallet_path`, `random_seed` and `num_topics` in this method, it will be good to define the signature as:

```
def train(data, mallet_path: Path, random_seed: int, num_topics: int, *args, **kwargs):
```

the invoke of this method would be the same as before.

Reply:
done